### PR TITLE
Add Sorbet 0.4.5085.20191130175112-e9dddf67d

### PR DESCRIPTION
**Pull request by fzakaria**
Added Sorbet https://sorbet.org Cask.
I had attempted to create a homebrew formula https://github.com/Homebrew/homebrew-core/pull/47173
however was unable to get the formula to build on all OSX versions.

Opted instead to create a cask which distributes the prebuilt binaries
by the sorbet team.

After making all changes to the cask:

- [x] `brew cask audit --download {{cask_file}}` is error-free.
```brew cask audit sorbet --download
==> Downloading https://github.com/sorbet/sorbet/releases/download/0.4.5085.20191130175112-e9dddf67d/mac.sorbet
Already downloaded: /Users/faridzakaria/Library/Caches/Homebrew/downloads/59f9f3451a944dafa6613811e8821d99dccc13acdf16696d3e3f0a2f0f8ad0f5--mac.sorbet
==> Verifying SHA-256 checksum for Cask 'sorbet'.
audit for sorbet: passed
```
- [x] `brew cask style --fix {{cask_file}}` reports no offenses.
```
$ brew cask style Casks/sorbet.rb     
1 file inspected, no offenses detected
```
- [x] The commit message includes the cask’s name and version.
- [x] The submission is for [a stable version](https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#finding-a-home-for-your-cask) or [documented exception](https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#but-there-is-no-stable-version).

Additionally, if **adding a new cask**:

- [x] Named the cask according to the [token reference].
- [x] `brew cask install {{cask_file}}` worked successfully.
- [x] `brew cask uninstall {{cask_file}}` worked successfully.
- [x] Checked there are no [open pull requests] for the same cask.
- [x] Checked the cask was not [already refused].
- [x] Checked the cask is submitted to [the correct repo].

[token reference]: https://github.com/Homebrew/homebrew-cask/blob/master/doc/cask_language_reference/token_reference.md
[open pull requests]: https://github.com/Homebrew/homebrew-cask/pulls
[already refused]: https://github.com/Homebrew/homebrew-cask/search?q=is%3Aclosed&type=Issues
[the correct repo]: https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#finding-a-home-for-your-cask
[version-checksum]: https://github.com/Homebrew/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/sha256.md#updating-the-sha256

### CODE REVIEW

**Comment by DarkDimius** [comment]
Maintainer of upstream sorbet here.
Is it possible to call it `sorbet-static`?

The reason being that this only includes the internal static binary and doesn't include the runtime component nor does it include the `srb` wrapper, thus making it very special and pretty much not covered by docs.

Calling it `sorbet-static` will also match the name that this binary has in rubygems: https://rubygems.org/gems/sorbet-static

**Comment by vitorgalvao** [comment]
@fzakaria Welcome, and thank you for the submission, but this isn’t how we work. Unless there’s an impossibility of adding this to homebrew-core we won’t accept it here, and so far it seems to be possible, even it hasn’t gotten there yet.

@DarkDimius 

> Is it possible to call it `sorbet-static`?

That complicates things, because from what you’re saying it seems like including these instead of compiling from source could lead to users not getting exactly what they want (cue bug reports on our side). Adding the `-static` would be confusing with our naming schemes.

Seems like the best bet for now would be if you both could work on getting the formula (i.e. https://github.com/Homebrew/homebrew-core/pull/47173) working. Having an upstream maintainer looking into it as well is bound to make it easier.

**Comment by DarkDimius** [comment]
> That complicates things...

> Seems like the best bet for now would be if you both could work on getting the formula (i.e. Homebrew/homebrew-core#47173) working.

Note that that pull request has the same issue: it only compiles the static library(and thus excludes `srb`, and thus most documentation examples won't work) and goes against common way to manage versions of dependencies in ruby(via Gemfile).

ooc: what's the motivation to include it in homebrew in the first place compared to using the documented gemfile installation?

**Comment by fzakaria** [comment]
Okay :/ 

<ooc>

@DarkDimius I described the problem statement in https://github.com/sorbet/sorbet/issues/2213

tl;dr; If you are using a non-standard ruby interpreter i.e. JRuby -- you cannot install sorbet-static via gem due to the platform restriction. The platform restriction is also non-traditional since sorbet-static delivers a separate utility vs. a library to be `required` in. 

I was hoping to make `sorbet` (sorbet-static) installable through other mens other than `gem` as I work on integration with our JRuby codebase. The binary is offered on the _release_ page which is great -- I was aiming for a more ergonomic installation.

</ooc>

**Comment by DarkDimius** [comment]
@fzakaria I see, would it be possible for you to join Sorbet slack to discuss options? https://sorbet.org/slack
